src/main/java/net/minestom/server/item/ItemTag.java

```diff
@@ -0,0 +1,97 @@
+import org.jglrxavpok.hephaistos.nbt.NBTCompound;
+
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+public class ItemTag<T> {
+
+    private final String key;
+    private final Function<NBTCompound, T> readFunction;
+    private final BiConsumer<NBTCompound, T> writeConsumer;
+
+    private ItemTag(@NotNull String key,
+                    @NotNull Function<NBTCompound, T> readFunction,
+                    @NotNull BiConsumer<NBTCompound, T> writeConsumer) {
+        this.key = key;
+        this.readFunction = readFunction;
+        this.writeConsumer = writeConsumer;
+    }
+
+    public @NotNull String getKey() {
+        return key;
+    }
+
+    protected T read(@NotNull NBTCompound nbtCompound) {
+        return readFunction.apply(nbtCompound);
+    }
+
+    protected void write(@NotNull NBTCompound nbtCompound, @NotNull T value) {
+        this.writeConsumer.accept(nbtCompound, value);
+    }
+
+    public static @NotNull ItemTag<Byte> Byte(@NotNull String key) {
+        return new ItemTag<>(key,
+                nbtCompound -> nbtCompound.getByte(key),
+                (nbtCompound, value) -> nbtCompound.setByte(key, value));
+    }
+
+    public static @NotNull ItemTag<Short> Short(@NotNull String key) {
+        return new ItemTag<>(key,
+                nbtCompound -> nbtCompound.getShort(key),
+                (nbtCompound, value) -> nbtCompound.setShort(key, value));
+    }
+
+    public static @NotNull ItemTag<Integer> Integer(@NotNull String key) {
+        return new ItemTag<>(key,
+                nbtCompound -> nbtCompound.getInt(key),
+                (nbtCompound, integer) -> nbtCompound.setInt(key, integer));
+    }
+
+    public static @NotNull ItemTag<Long> Long(@NotNull String key) {
+        return new ItemTag<>(key,
+                nbtCompound -> nbtCompound.getLong(key),
+                (nbtCompound, value) -> nbtCompound.setLong(key, value));
+    }
+
+    public static @NotNull ItemTag<Float> Float(@NotNull String key) {
+        return new ItemTag<>(key,
+                nbtCompound -> nbtCompound.getFloat(key),
+                (nbtCompound, value) -> nbtCompound.setFloat(key, value));
+    }
+
+    public static @NotNull ItemTag<Double> Double(@NotNull String key) {
+        return new ItemTag<>(key,
+                nbtCompound -> nbtCompound.getDouble(key),
+                (nbtCompound, value) -> nbtCompound.setDouble(key, value));
+    }
+
+    public static @NotNull ItemTag<byte[]> ByteArray(@NotNull String key) {
+        return new ItemTag<>(key,
+                nbtCompound -> nbtCompound.getByteArray(key),
+                (nbtCompound, value) -> nbtCompound.setByteArray(key, value));
+    }
+
+    public static @NotNull ItemTag<String> String(@NotNull String key) {
+        return new ItemTag<>(key,
+                nbtCompound -> nbtCompound.getString(key),
+                (nbtCompound, value) -> nbtCompound.setString(key, value));
+    }
+
+    // TODO List/Compound
```

Comment:
Perhaps one for `Component`?

Reply:
There should only be methods for official NBT types https://wiki.vg/NBT#Specification
`Component` can be serialized to string as a workaround